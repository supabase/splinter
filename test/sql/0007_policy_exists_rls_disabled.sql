begin;
  set local search_path = '';

  create table public.blog(
    id int primary key
  );

  create policy select_own_posts on public.blog
  for select
  using (true);

  create policy all_own_posts on public.blog
  for all
  using (true);

  -- 1 issue
  select * from lint."0007_policy_exists_rls_disabled";

  -- resolve the issue
  alter table public.blog enable row level security;

  -- 0 issue
  select * from lint."0007_policy_exists_rls_disabled";

rollback;
