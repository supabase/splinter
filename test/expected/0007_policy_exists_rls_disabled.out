begin;
  set local search_path = '';
  create table public.blog(
    id int primary key
  );
  create policy select_own_posts on public.blog
  for select
  using (true);
  create policy all_own_posts on public.blog
  for all
  using (true);
  -- 1 issue
  select * from lint."0007_policy_exists_rls_disabled";
            name            | level |  facing  | categories |                                                           description                                                           |                                                             detail                                                             |                                          remediation                                           |                       metadata                        |               cache_key                
----------------------------+-------+----------+------------+---------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------+-------------------------------------------------------+----------------------------------------
 policy_exists_rls_disabled | INFO  | EXTERNAL | {SECURITY} | Detects cases where row level security (RLS) policies have been created, but RLS has not been enabled for the underlying table. | Table \`public.blog\` has RLS policies but RLS is not enabled on the table. Policies include {all_own_posts,select_own_posts}. | https://supabase.com/docs/guides/database/database-linter?lint=0007_policy_exists_rls_disabled | {"name": "blog", "type": "table", "schema": "public"} | policy_exists_rls_disabled_public_blog
(1 row)

  -- resolve the issue
  alter table public.blog enable row level security;
  -- 0 issue
  select * from lint."0007_policy_exists_rls_disabled";
 name | level | facing | categories | description | detail | remediation | metadata | cache_key 
------+-------+--------+------------+-------------+--------+-------------+----------+-----------
(0 rows)

rollback;
